### DISCUSSION TITLE
about the chat history in client

### DISCUSSION THREAD

**Discussion by cbdgftqs** [LiveHelperChat/livehelperchat#1824] [General]
Hi:
is there any way that the client can retain the chat history for the logged user with uuid, if not can they preserve their chat log?

**Comment by remdex** [comment]
In https://doc.livehelperchat.com/docs/javascript-arguments there you can set your own UUID. So online visitor record will be the same always.

For operator
Operator always will see his previous chats.

As for visitor
After that you can just enable show previous chats for the same vistor. https://doc.livehelperchat.com/docs/theme/#show-previous-chat-messages-in-chat-widget

**Reply by cbdgftqs** [reply]
Great! it is so powerful, Thanks a lot!
